src/runtime/contrib/tensorrt/tensorrt_builder.cc

```diff
@@ -208,8 +208,35 @@ runtime::TrtEngineAndContext TensorRTBuilder::BuildEngine(
   CHECK_EQ(engine->getNbBindings(),
            num_input_bindings + network_output_names_.size());
   nvinfer1::IExecutionContext* context = engine->createExecutionContext();
-  return {engine, context, network_input_names_, network_input_is_baked_,
-          network_output_names_};
+  std::vector<runtime::NDArray> device_buffers;
+  for (int i = 0; i < network_input_names_.size(); i++) {
+    if (network_input_is_baked_[i]) continue;
+    int binding_index = engine->getBindingIndex(network_input_names_[i].c_str());
+    if (execution_args_[binding_index]->ctx.device_type == kDLGPU) {
+      device_buffers.emplace_back();
+    } else {
+      std::vector<int64_t> shape_(
+          execution_args_[binding_index]->shape,
+          execution_args_[binding_index]->shape + execution_args_[binding_index]->ndim);
+      device_buffers.push_back(
+          runtime::NDArray::Empty(shape_, execution_args_[binding_index]->dtype, {kDLGPU, 0}));
+    }
+  }
+  for (int i = 0; i < network_output_names_.size(); i++) {
+    int binding_index = engine->getBindingIndex(network_output_names_[i].c_str());
+    if (execution_args_[binding_index]->ctx.device_type == kDLGPU) {
```

Comment:
For all indexes into execution args, use 
`const int index_in_args = execution_args_.size() - network_output_names_.size() + i`